packages/geo/src/tms/tms.loader.ts

```diff
@@ -0,0 +1,158 @@
+import { getXyOrder, GoogleTms, Projection, ProjectionLoader, TileMatrixSet, TileMatrixSetType } from '../index.js';
+
+/**
+ * Minimal typings of PROJJSON
+ */
+interface ProjJsonAxis {
+  name: string;
+  abbreviation: 'x' | 'y' | 'e' | 'n';
+  direction: 'east' | 'north';
+  unit: string;
+}
+
+interface ProjJson {
+  name: string;
+  bbox: {
+    south_latitude: number;
+    west_longitude: number;
+    north_latitude: number;
+    east_longitude: number;
+  };
+  coordinate_system: {
+    subtype: 'Cartesian';
+    axis: [ProjJsonAxis, ProjJsonAxis];
+  };
+}
+/**
+ * Find a zoom level that has approximately the same bounds as those of the source projection.
+ * The WebMercatorQuad projection covers the entire world. So, most other projections do not work well outside
+ * their own bounds. Therefore, we want to find a zoom level that does not overflow the bounds too much.
+ *
+ * @param minSize
+ * @returns one zoom level bigger than the minimum zoom level
+ */
+function findZoomOffset(minSize: number): number {
+  for (let i = 1; i < GoogleTms.maxZoom; i++) {
+    const size = GoogleTms.pixelScale(i) * 256;
+    const sizeDiff = minSize - size;
+    if (sizeDiff > 0) return Math.max(0, i - 1);
+  }
+  return 0;
+}
+
+/**
+ * Normalize axis names to X or Y
+ *
+ * @param ax Axis
+ * @returns Normalized axis name
+ * @throws if axis is unknown
+ */
+function axisName(ax: ProjJsonAxis): 'x' | 'y' {
+  switch (ax.abbreviation.toLowerCase()) {
+    case 'x':
+    case 'e':
+      return 'x';
+    case 'y':
+    case 'n':
+      return 'y';
+    default:
+      throw new Error('Unknown axis abbreviation: ' + ax.abbreviation);
+  }
+}
+
+export class TmsLoader {
+  /**
+   * Attempts to generate a runtime Tile Matrix Set for the given target projection.
+   *
+   * - Lookup the projection information from spatial reference
+   * - Attempt to find the bounds of the projection
+   * - Using the bounds, attempt to find a tile matrix zoom level that would cover the entire bounds
+   * - Generate a runtime tile matrix definition
+   *
+   * @param epsgCode - An Epsg code expressing the target projection (e.g. 3788, 5479, or 32702)
+   * @returns the generated tile matrix definition
+   */
+  static async load(epsgCode: number): Promise<TileMatrixSet> {
+    // load the epsg code's projection definition
+    const epsg = await ProjectionLoader.load(epsgCode);
+    const proj = Projection.get(epsg);
+
+    // fetch projection metadata from the spatialreference.org API,
+    // this givens us valid axis information and geographic bounding box
+    const projJson = (await fetch(`https://spatialreference.org/ref/epsg/${epsg.code}/projjson.json`).then((f) =>
```

Comment:
this could add a ton of load on this server, I wonder if we need to store these somewhere else, as for every tile lambda this url could be hit?

Reply:
This needs to happen per lambda for request, this is still going to be a ton of load on this third party server, and also add them to our dependency chain, (you cannot serve a tile for Antarctica without hitting this server first)

I think we need to store the common projjson's for all of our expected projections, these could just be in a `tms/proj/2193.ts` or similar.

We should not need to fetch a external URL when trying to serve a tile.